pkg/servicemesh/control-plane.go

```diff
@@ -0,0 +1,493 @@
+
+			builder.errorMsg = "The Grafana addon 'address' cannot be empty when Grafana addon is enabled"
+		}
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	addons := &istioV2.AddonsConfig{
+		Grafana: &istioV2.GrafanaAddonConfig{
+			Enablement: istioV2.Enablement{
+				Enabled: &enablement,
+			},
+			Install: grafanaInstallConfig,
+			Address: &address,
+		},
+	}
+
+	builder.Definition.Spec.Addons = addons
+
+	return builder
+}
+
+// WithJaegerAddon adds joeger addon to the serviceMeshControlPlane.
+func (builder *ControlPlaneBuilder) WithJaegerAddon(
+	name string,
+	jaegerInstallConfig *istioV2.JaegerInstallConfig,
+) *ControlPlaneBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating serviceMeshControlPlane %s with the JaegerAddonConfig defined", builder.Definition.Name)
+
+	if name == "" {
+		glog.V(100).Infof("The name of the Jaeger addon is empty")
+
+		builder.errorMsg = "The Jaeger addon 'name' cannot be empty"
+	}
+
+	if jaegerInstallConfig == nil {
+		glog.V(100).Infof("The jaegerInstallConfig of the Jaeger addon is empty")
+
+		builder.errorMsg = "The Jaeger addon 'jaegerInstallConfig' cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	addons := &istioV2.AddonsConfig{
+		Jaeger: &istioV2.JaegerAddonConfig{
+			Name:    name,
+			Install: jaegerInstallConfig,
+		},
+	}
+
+	builder.Definition.Spec.Addons = addons
+
+	return builder
+}
+
+// WithKialiAddon adds kiali addons to the serviceMeshControlPlane.
+func (builder *ControlPlaneBuilder) WithKialiAddon(
+	enablement bool,
+	name string,
+	kialiInstallConfig *istioV2.KialiInstallConfig) *ControlPlaneBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating serviceMeshControlPlane %s with the KialiAddonConfig defined", builder.Definition.Name)
+
+	if enablement {
+		if kialiInstallConfig == nil {
+			glog.V(100).Infof("The kialiInstallConfig of the Kiali addon is empty")
+
+			builder.errorMsg = "The Kiali addon 'kialiInstallConfig' cannot " +
+				"be empty when Kiali addon is enabled"
+		}
+
+		if name == "" {
+			glog.V(100).Infof("The name of the Kiali addon is empty")
+
+			builder.errorMsg = "The Kiali addon 'name' cannot be empty when Kiali addon is enabled"
+		}
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	addons := &istioV2.AddonsConfig{
+		Kiali: &istioV2.KialiAddonConfig{
+			Enablement: istioV2.Enablement{
+				Enabled: &enablement,
+			},
+			Name:    name,
+			Install: kialiInstallConfig,
+		},
+	}
+
+	builder.Definition.Spec.Addons = addons
+
+	return builder
+}
+
+// WithPrometheusAddon adds prometheus addons to the serviceMeshControlPlane.
+func (builder *ControlPlaneBuilder) WithPrometheusAddon(
+	enablement bool,
+	metricsExpiryDuration string,
+	scrape bool,
+	prometheusInstallConfig *istioV2.PrometheusInstallConfig,
+	address string,
```

Comment:
Please keep argument of the same type close to each other. Move address var after > metricsExpiryDuration string